tests/plugins/plugin.text_selection.test.js

```diff
@@ -0,0 +1,102 @@
+import sinon from 'sinon';
+import '../../BookReader/jquery-1.10.1.js';
+import '../../BookReader/jquery-ui-1.12.0.min.js';
+import '../../BookReader/jquery.browser.min.js';
+import '../../BookReader/dragscrollable-br.js';
+import '../../BookReader/jquery.colorbox-min.js';
+
+import '../../src/js/BookReader.js';
+import BookreaderWithTextSelection from '../../src/js/plugins/plugin.text_selection.js';
+import { PageModel, BookModel } from '../../src/js/BookReader/BookModel.js';
+
+/** @type {BookReader} */
+const FAKE_XML = `<OBJECT data="file://localhost//tmp/derive/goodytwoshoes00newyiala//goodytwoshoes00newyiala.djvu" height="3192" type="image/x.djvu" usemap="goodytwoshoes00newyiala_0001.djvu" width="2454">
+<PARAGRAPH><LINE><WORD coords="1216,2768,1256,2640">test</WORD></LINE></PARAGRAPH></OBJECT>`;
+let br;
+const LONG_PRESS_DURATION = 500;
+
+beforeEach(() => {
+  const parser = new DOMParser()
+  document.body.innerHTML = '<div id="BookReader">';
+  br = new BookreaderWithTextSelection({
+    data: [
+      [
+        { width: 800, height: 1200,
+          uri: '//archive.org/download/BookReader/img/page001.jpg' },
+      ],
+      [
+        { width: 800, height: 1200,
+          uri: '//archive.org/download/BookReader/img/page002.jpg' },
+        { width: 800, height: 1200,
+          uri: '//archive.org/download/BookReader/img/page003.jpg' },
+      ],
+      [
+        { width: 800, height: 1200,
+          uri: '//archive.org/download/BookReader/img/page004.jpg' },
+        { width: 800, height: 1200,
+          uri: '//archive.org/download/BookReader/img/page005.jpg' },
+      ]
+    ],
+  });
+  br.enableTextSelection = true;
+  br.init();
+  sinon.stub(br.textSelectionPlugin, "getPageText").callsFake(async (pageIndex) => {
+    return $(parser.parseFromString(FAKE_XML, "text/xml"));
+  });
```

Comment:
```suggestion
  sinon.stub(br.textSelectionPlugin, "getPageText")
      .resolves($(parser.parseFromString(FAKE_XML, "text/xml")));
```

Reply:
pageIndex param is worth to keep because it's used to choose between different xml elements